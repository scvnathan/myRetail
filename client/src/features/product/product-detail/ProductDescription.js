import React from "react";
import styled from "styled-components";

const List = styled.ul`
	margin: 0 0 0 1em;
	padding: 0;
	color: #777;
`;

const Item = styled.li`
	color: #777;
	margin-bottom: 0.5em;
`;

const Highlights = styled.h1`
	font-weight: normal;
`;

export default props => {
	const features = props.features;
	return (
		<div>
			<Highlights>product highlights</Highlights>
			<List>
				{features.map(feature => {
					return <Item key={feature}>{feature}</Item>;
				})}
			</List>
		</div>
	);
};
