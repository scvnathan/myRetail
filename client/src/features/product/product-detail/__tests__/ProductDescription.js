import React from "react";
import ProductDescription from "~root/features/product/product-detail/ProductDescription"
import renderer from "react-test-renderer";

describe("ProductDescription component", () => {
  it("Test render correctly", () => {
    const comp = <ProductDescription features={["Feature 1", "Feature 2"]}/>
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });
});