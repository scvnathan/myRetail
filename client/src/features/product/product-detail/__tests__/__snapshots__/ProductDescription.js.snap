// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDescription component Test render correctly 1`] = `
<div>
  <h1
    className="ProductDescription__Highlights-sc-1xj5pwi-2 fgyiTF"
  >
    product highlights
  </h1>
  <ul
    className="ProductDescription__List-sc-1xj5pwi-0 gnVBWn"
  >
    <li
      className="ProductDescription__Item-sc-1xj5pwi-1 AFuDf"
    >
      Feature 1
    </li>
    <li
      className="ProductDescription__Item-sc-1xj5pwi-1 AFuDf"
    >
      Feature 2
    </li>
  </ul>
</div>
`;
